### ISSUE TITLE
Need more documentation

### ISSUE THREAD

**Issue by TheZenKitteh** [Byzantium/Byzantium#224]
I am totally new to this project and fairly new to wireless meshes. I managed to get Byzantium on a USB stick, running and working. However I can't find any documentation on how to tweak the setup or add more services (like say game servers or file servers or whatever). I also can't seem to find the location of the file that shows when a user first connects to the node (the page that lists the available services). I did manage to hunt down a file and changed the ESSID of the node, but afraid to tweak other settings without more information on what they do (and if it will break anything). Basically there needs to be a lot more documents. I wish I could help with such but I don't know enough about the applications involved.

EDIT:
- Found the location of the page that lists the available services. /srv/httpd/htdocs it seems.

**Comment by virtadpt** [comment]
You're right, this is something that we really have to document.  Let me see...

Tweaking the setup when you've installled Byzantium Linux to a USB key is pretty easy assuming that you've got persistent changes enabled (with a storage file or if the drive is formatted with a Linux filesystem).  Edit the config files and initscripts the way you'd admin any other Linux machine, and the changes will be saved to the USB key.

There are two ways to add new services: Remaster the Byzantium Linux .iso image, or install them on your USB key.  The latter is pretty straightforward: Boot your Byzantium USB key, connect to the Net, download what you need (dependencies, source code, what have you), and install them per their directions as if you were using a less specialized distro of Linux running on a desktop or laptop.  If you have persistent storage enabled they'll go their proper locations and the changes will be merged into the running system at boot-time when the persistent storage is activated.  If you were going to remaster the ISO image, you would use a process similar to this one: https://github.com/Byzantium/Byzantium/wiki/Process-for-manually-building-Byzantium-Linux

In short, we create a fakeroot directory inside a Porteus Linux VM, unpack all of the additional packages into it, copy our own code into it, and then create an .xzm file, which is a compressed overlay file system.  That .xzm file is mounted and spliced into the running system when activated (either at boot time, or by hand).

As for reconfiguring the wireless stack, editing byzantium_configd.py is sufficient.  The risk of doing so is that you'll lose compatibility with other Byzantium nodes because they all use the same BSSID and channel.  Changing the ESSID should not cause problems because ESSIDs are logical separations between networks and not physical ones.
